web/app/dao/DatasetsDAO.java

```diff
@@ -1806,14 +1807,14 @@ public static ReturnCode updateDatasetOwners(int datasetId, Map<String, String[]
 				owner.sortId = i;
 				owners.add(owner);
 
-				if (StringUtils.isNotBlank(confirmedBy)) {
-					confirmedCount++;
+				if (type.equalsIgnoreCase("owner") && idType.equalsIgnoreCase("user") && StringUtils.isNotBlank(confirmedBy)) {
+					confirmedOwnerUserCount++;
 				}
 			}
 		}
 
     // enforce at least two confirmed owner for a dataset before making any changes
-		if (confirmedCount < 2) {
+		if (confirmedOwnerUserCount < 2) {
```

Comment:
Due to time constraint, we will leave this as a fix number for now. Since a change in the play config will also require changes in the multiproduct scripts. We can do this later.